File: Leetcode/Python/_520.py
class Solution:
    def detectCapitalUse(self, word: str) -> bool:
        capitals = 0
        for char in word:
            if char.isupper():
                capitals += 1
        if capitals == 1 and word[0].isupper():
            return True
        elif capitals == 0:
            return True
        elif capitals == len(word):
            return True
        else:
            return False
        
class Solution:
    def detectCapitalUse(self, word: str) -> bool:
        caps = False
        low = False
        first = False
        for i, c in enumerate(word):
            if i == 0:
                if c.isupper():
                    first = True
            else:
                if c.isupper():
                    caps = True
                else:
                    low = True
        if (low is True and caps is False and first is True) or (first is False and low is True and caps is False) or (first is True and low is False and caps is True) or len(word) == 1:
            return True
        return False

File: Leetcode/cpp/520.cpp
#include <iostream>
#include <string>
#include <cassert>

class Solution {
public:
    bool detectCapitalUse(std::string word) {
        int n = word.length();
        int first = 0;
        int all = 0;
        int lower = 0;
        for (int i = 0; i < n; ++i) {
            char ch = word[i];
            if (i == 0) {
                if (ch >= 'A' && ch <= 'Z') {
                    first = 1;
                }
            }
            if (ch >= 'A' && ch <= 'Z') {
                ++all;
            } else {
                ++lower;
            }
        }
        if (all == n || (first && ((lower + 1) == n)) || lower == n) {
            return true;
        }
        return false;
    }
};

int main() {
    Solution sol;
    std::string word = "FlaG";
    bool expected_output = false;
    bool output = sol.detectCapitalUse(word);
    assert(output == expected_output && "Test #1 failed!");

    std::cout << "Tests are passed!\n";
    return 0;
}
